homeassistant/components/plant.py

```diff
@@ -206,14 +231,65 @@ def _update_state(self):
 
         if result:
             self._state = STATE_PROBLEM
-            self._problems = ','.join(result)
+            self._problems = ', '.join(result)
         else:
             self._state = STATE_OK
             self._icon = 'mdi:thumb-up'
             self._problems = PROBLEM_NONE
         _LOGGER.debug("New data processed")
         self.async_schedule_update_ha_state()
 
+    def check_brightness(self):
+        """
+        Check brightness levels over a history of several days.
+
+        It usually does not make sense to check the minimum brightness now as
+        it might be night. So we need to check this over several days to see if
+        one of those days was bright enough.
+
+        As this operation is quite expensive, we should not run it so
+        frequently. Thus we only check when the brightness really changed
+        (ie self._brightness_updated == True). The result of the last check
+        is cached in self._brightness_problem.
+        """
+        from homeassistant.components.recorder.models import States
```

Comment:
Look into history sensor how you should do that right. Short: Read that only on startup and store it into https://docs.python.org/2/library/collections.html#collections.deque. Allow user to choise the size of this. Now track the sensore state change and write it into this queue.

Reply:
While I understand your proposal, I'm not really happy with it:
1. I'm re-implementing functionality that is already implemented in the database: Store history of a value, get last 3 days of measurements.
1. While it reduces the database read operations, you spend additional memory to store the data. It's the classical trade-off between CPU time and RAM. In this case I prefer the database.
1. I do not care how many samples there are, I care about the last N days. So the queue might get quite long, if you have a high sampling rate on your sensor. To avoid that I would have to implement some smart way of compressin/downsampling the data. And that's just overkill for such a simple feature.

So all in all, I think the current solution is still the best option.

Reply:
Took me quite some time, but now the data is compressed, so that we're storing only one entry per day. That's enough for this use case as we only want to know the maximum brightness of the course of a day.

And the data is read from the database only on startup, not during normal operation. And only if the recorder is available, so this is optional.